### PR TITLE
DBO-88, DBO-91 - Censoring Bad Words and New Flagged Field

### PR DESCRIPTION
**Pull request by nathanielcaron**
`DBO-88 Censor words from user names, post titles, and post bodies in the create post endpoints.`
`DBO-91 Add 'Flagged' boolean field in post document`

- Added `bad-words` package
- Created `isStringProfane()` and `cleanString()` util functions and unit tests
- Made use of util functions to clean user names, post titles, and post bodies
- Added some Karate API tests with profane language

### CODE REVIEW

File: Service/db/dbUtils.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ require('dotenv').config();
 const ENV = process.env;
 
 module.exports.connectTest = async (name) => {
+  // eslint-disable-next-line max-len
```
Comment (sahilsaini432) [inline Service/db/dbUtils.js:6]: NIT: You can add this to the eslint ignore file.

Comment (nathanielcaron) [reply Service/db/dbUtils.js:6]: I personally think we should have a max line length. I set it to 120 characters.

**Review by fredy20501 — APPROVED** [review]
Really ******* amazing! 🥇

**Review by sahilsaini432 — APPROVED** [review]
💯

**Review by CarterMacLennan — APPROVED** [review]

**Review by JoaoPontes98 — APPROVED** [review]

**Pull request merged**